core/src/main/java/org/fao/geonet/kernel/DataManager.java

```diff
@@ -117,14 +117,14 @@ public void init(ServiceContext context, Boolean force) throws Exception {
         this.metadataUtils.init(context, force);
 
         // FIXME this shouldn't login automatically ever!
-        if (context.getUserSession() == null) {
-            LOGGER_DATA_MANAGER.debug( "Automatically login in as Administrator. Who is this? Who is calling this?");
-            UserSession session = new UserSession();
-            context.setUserSession(session);
-            session.loginAs(new User().setUsername("admin").setId(-1).setProfile(Profile.Administrator));
-            LOGGER_DATA_MANAGER.debug( "Hopefully this is cron job or routinely background task. Who called us?",
-                        new Exception("Dummy Exception to know the stacktrace"));
-        }
+//        if (context.getUserSession() == null) {
+//            LOGGER_DATA_MANAGER.debug( "Automatically login in as Administrator. Who is this? Who is calling this?");
+//            UserSession session = new UserSession();
+//            context.setUserSession(session);
+//            session.loginAs(new User().setUsername("admin").setId(-1).setProfile(Profile.Administrator));
+//            LOGGER_DATA_MANAGER.debug( "Hopefully this is cron job or routinely background task. Who called us?",
+//                        new Exception("Dummy Exception to know the stacktrace"));
+//        }
```

Comment:
Okay the above comment cried out ``// FIXME this shouldn't login automatically ever!`` so I am taking this opportunity to answer.

This change change may be seen in two ways:

a) other sections of the code (monitors, harvesters, batch process, bulk edit) may of accidentally gotten admin permissions before! So if we see some subtle permission differences we will know what happened...

b) other code using the shared appHandler service context may wish to assume a user session and be upset when one is not available at all (null pointer exception etc...)